evaluation/evaluate.py

```diff
@@ -194,81 +196,224 @@ def run_evaluation_from_config(working_dir: Path, config: dict, num_questions: i
         logger.error("Evaluation was terminated early due to an error ⬆")
 
 
+def plot_bar_charts(
+    layout: Tuple[int, int],
+    data: List[Dict[str, any]],
+    titles: List[str],
+    y_labels: List[str],
+    y_max_lim: List[float] = None,
+    width=0.4,
+    output_path: Path = Path("evaluation_results.png"),
+):
+    """Plot bar charts for the provided data."""
+    assert layout[0] * layout[1] == len(data), "Number of data points must match the layout"
+
+    font = {"weight": "normal", "size": 9}
+
+    matplotlib.rc("font", **font)
+
+    fig, axs = plt.subplots(layout[0], layout[1], figsize=(layout[1] * 5, layout[0] * 4))
+    fig.tight_layout(pad=3.0)
+
+    for i, ax in enumerate(axs.flat):
+        x_data = list(data[i].keys())
+        y_data = list(data[i].values())
+        name = titles[i]
+        colors = plt.cm.tab20.colors[: len(x_data)]
+
+        ax.bar(x_data, y_data, width=width, label=name, color=colors)
+        ax.set_title(name, pad=20)
+        if y_max_lim is not None and len(y_max_lim) > i and y_max_lim[i] is not None:
+            ax.set_ylim(0, max(y_max_lim[i], max(y_data)))
+        else:
+            ax.set_ylim(0, np.ceil(max(y_data)) * 1.2)
+
+        ax.set_ylabel(y_labels[i])
+
+        for j, v in enumerate(y_data):
+            ax.text(j, v * 1.02, str(round(v, 2)), ha="center")
+
+    plt.savefig(output_path)
+    plt.close(fig)
+
+
+def plot_box_charts(
```

Comment:
Why do we need both this and `plot_box_chart`? Can't we abstract the box chart creation in a single function and control any sort of special behaviours through params?

Reply:
Because it looks like this one just executes a more elaborate version of `plot_box_chart`.

Reply:
Well, we *could* use `plot_box_charts` to plot a 1x1 layout with only one group of data. But it makes function call ugly and the parameter inputs less readable. I still insist on making them two distinct functions: one `plot_box_charts` that plots multiple box plot graphs for one set of data each, and another one `plot_box_chart` that plots one box plot graph for multiple sets of data.

Reply:
@GrayNekoBean I wouldn't mind it too much if `plot_box_charts` at least called the other one. This way it's just weird given that they practically have the same name. If one of them does sth super particular, perhaps the name should portray that. It currently reads "call me to plot any a basic box chart with the input data" and "call me to plot multiple basic box charts with the input data".

Reply:
Once again, note that these will be extended very shortly to allow for comparing LLMs, so they should be written in a flexible way.